### ISSUE TITLE
Content warning should hide (or blur) images as well

### ISSUE THREAD

**Issue by schmaker** [mastodon/mastodon#25732]
### Steps to reproduce the problem

Make post (tried from Friendica, not sure if it's same within Mastodon) with content warning and single image. On other networks, image is hidden, Mastodon shows it (not even blur is used).

### Expected behaviour

Image should be hidden or blurred

### Actual behaviour

Image is shown

### Detailed description

In Friendica, image is hidden (and it works same way in Misskey)
![image](https://github.com/mastodon/mastodon/assets/14954032/4bce784a-188a-4a57-b3e3-11a1db93d5b0)
in Mastodon only text is hidden and image is not even blurred
![image](https://github.com/mastodon/mastodon/assets/14954032/9af353cd-4fe1-4fba-bb03-6b9b6812dd7b)

I think it's safe to assume, that when anybody posts something with CW, it's linked images are also "dangerous" content.

I'm not sure if it's federation issue or general one as i am not actually Mastodon user

### Specifications

Receiving instance is running Mastodon 4.0.2

**Comment by renchap** [comment]
On Mastodon, when you make a post with a Content Warning, the post is marked as `sensitive` automatically, which hides the image.

I assume this is caused by Friendica adding a Content Warning to the post, but not marking it as `sensitive`.

**Comment by schmaker** [comment]
OK, let's assume that it means that it is issue on sender-side federation server.

Does not this also means that it's easily exploitable by malicious servers? I still do think that it should also be handled on Mastodon receiver-side server in same way that Friendica and Misskey does - hide before clicking on CW

**Comment by renchap** [comment]
Yes, it could be a good idea on the Mastodon's side to set the `sentitive` property if the incoming post has a Content Warning.

**Comment by JupiterRowland** [comment]
The issue here lies in Mastodon's treatment of images coming in from outside of Mastodon.

Images uploaded to Mastodon itself and then attached to a Mastodon post are being blurred if there's a content warning on the post.

Images coming in from certain other projects, especially if they handle images differently, are shown in plain sight, even if there's what Mastodon understands as a content warning on the post.

Friendica, Hubzilla and (streams) do handle images differently. On these projects, images aren't uploaded to "somewhere" and added to posts as file attachments. Instead, they're uploaded to the manageable file space that belongs to the account/channel. Then they're embedded in the posts as links to where the image file is uploaded. It works the same as on a blog or in a forum.

Mastodon doesn't understand linked images. So in order to preserve them (or at least the last four if there are more than four images linked into a post), it converts these linked images into file attachments. However, even with a summary (Friendica, Hubzilla, (streams))/content warning (Mastodon) on the post, these attached images are not being marked as sensitive and therefore blurred.

This renders content warnings half-useless for users of just about everything that isn't Mastodon because they can't be sure if a summary/content warning will blur their images on Mastodon. Essentially, non-Mastodon users could barely post images at all if they happen to contain something sensitive.

By the way, Friendica, Hubzilla and (streams) have their own way of marking images sensitive. This may be off-topic, and this may require changes on their sides, but it'd be nice if Mastodon understood these.

**Comment by JupiterRowland** [comment]
If you need proof, [search a major instance like mastodon.social for the hashtag #osg16b](https://mastodon.social/tags/osg16b). Then you'll come across [a post from Hubzilla](https://mastodon.social/@jupiter_rowland@hub.netzgemeinde.eu/110799547446166542) (this link will most likely take you to Hubzilla to see the original as a proof that this post does come from Hubzilla) with a content warning on it and an unblurred, clearly visible image below.

**Comment by tsargoth** [comment]
I'd like to up vote this issue. I'm experiencing the same behaviour as well.
